Source/FakeItEasy/Core/ArgumentValueFormatter.cs

```diff
@@ -31,11 +32,8 @@ public virtual string GetArgumentValueAsString(object argumentValue)
             var argumentType = argumentValue.GetType();
 
             IArgumentValueFormatter formatter;
-            if (!this.cachedFormatters.TryGetValue(argumentType, out formatter))
-            {
-                formatter = this.ResolveTypeFormatter(argumentType);
-                this.cachedFormatters.Add(argumentType, formatter);
-            }
+            formatter = this.cachedFormatters.GetOrAdd(argumentType, (argType) =>
+                this.ResolveTypeFormatter(argType));

```

Comment:
The change looks technically sound, but has introduced some StyleCop warnings which could be cleared up by moving some arguments around. However, for my money, a superior formulation would be 

`var formatter = this.cachedFormatters.GetOrAdd(argumentType, this.ResolveTypeFormatter);`